src/NuGet.Core/NuGet.Protocol/HttpSource/HttpHandlerResourceV3Provider.cs

```diff
@@ -56,6 +58,11 @@ private HttpHandlerResourceV3 CreateResource(PackageSource packageSource)
                 AutomaticDecompression = (DecompressionMethods.GZip | DecompressionMethods.Deflate),
             };
 
+            if (packageSource.DisableTLSCertificateValidation)
+            {
+                clientHandler.ServerCertificateCustomValidationCallback = (HttpRequestMessage message, X509Certificate2 cert, X509Chain chain, SslPolicyErrors errors) => true;
```

Comment:
As I have mentioned in https://github.com/NuGet/Home/issues/4387#issuecomment-1186028492 comment, I think we should leverage [HttpClientHandler.DangerousAcceptAnyServerCertificateValidator](https://docs.microsoft.com/dotnet/api/system.net.http.httpclienthandler.dangerousacceptanyservercertificatevalidator?view=net-6.0#remarks) Property as suggested in the docs.

[Remarks](https://learn.microsoft.com/dotnet/api/system.net.http.httpclienthandler.dangerousacceptanyservercertificatevalidator?view=net-6.0#remarks) suggest that, 

Particularly in test scenarios, a common pattern use [HttpClient](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=net-6.0) to connect to a server with a certificate that shouldn't be validated, such as a self-signed certificate. You commonly do this with [HttpClientHandler](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler?view=net-6.0) by setting the [ServerCertificateCustomValidationCallback](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler.servercertificatecustomvalidationcallback?view=net-6.0) property to a delegate that always returns True; this indicates that the certificate has passed validation. However, not all implementations support this callback, and some throw [PlatformNotSupportedException](https://learn.microsoft.com/en-us/dotnet/api/system.platformnotsupportedexception?view=net-6.0).

The [DangerousAcceptAnyServerCertificateValidator](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler.dangerousacceptanyservercertificatevalidator?view=net-6.0) property addresses this limitation. The delegate returned by the [DangerousAcceptAnyServerCertificateValidator](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler.dangerousacceptanyservercertificatevalidator?view=net-6.0) property can be assigned to the [ServerCertificateCustomValidationCallback](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler.servercertificatecustomvalidationcallback?view=net-6.0) property

Reply:
`HttpClientHandler.DangerousAcceptAnyServerCertificateValidator` property is not available in netstandard2.0. After my discussion with Kartheek today, it was suggested that we could conditionally use `HttpClientHandler.DangerousAcceptAnyServerCertificateValidator` property based on the target framework. For netstandard2.0, we could use this code https://github.com/dotnet/runtime/blob/7532f65c9d8de91600ab5f36000295b0334e75d4/src/libraries/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs#L363 by copying the DangerousAcceptAnyServerCertificateValidator property implementation.

Reply:
Unless I'm missing something, the code you linked is just a complex way of returning `true`, except it doesn't increase the binary size when tree-shaking is used. NuGet isn't a core library where we care about tree-shaking optimization, so I don't see the benefit of copying that code.

Reply:
Make sense Andy. So, @Nigusu-Allehu my understanding is that for `netstandard2.0` we will just create a simple delegate that returns `true` similar to what you have right now. We rely on `HttpClientHandler.DangerousAcceptAnyServerCertificateValidator` for other target frameworks because it is recommended as per docs (I linked in my first comment on this thread)